### PR TITLE
Decaf + Rebrand + Updates

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR does expand it's scope slightly from the original manual decaf work, since the entirety of this repo is made up of just under 60 lines of code, with very few changes needed for the manual decaf work.

So in all this PR:
  * Decafs the source code
  * Removes all CoffeeScript based dependencies
  * Removes all Grunt logic, simplifying the way tests are run within `scripts` 
  * Rebrands the README and `package.json` to Pulsar
  * Removes useless, outdated files
  * Adds GitHub Action tests
  * Bumps the version from `1.1.3` => `1.1.4` to allow easy tagging once this is merged

These changes originate from the fact that very soon I am planning to update our usage of `property-accessors` everywhere we use it, so that we can remove our usage of `es5-ext` since older versions of `property-accessors` had used it, so if being updated everywhere, might as well get in these changes now

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,4 @@
-# Property Accessors Mixin
-[![OS X Build Status](https://travis-ci.org/atom/property-accessors.svg?branch=master)](https://travis-ci.org/atom/property-accessors)
-[![Windows Build Status](https://ci.appveyor.com/api/projects/status/ww4d10hi4v5h7kbp/branch/master?svg=true)](https://ci.appveyor.com/project/Atom/property-accessors/branch/master)
-[![Dependency Status](https://david-dm.org/atom/property-accessors.svg)](https://david-dm.org/atom/property-accessors)
+# Property Accessors Mixin [![CI](https://github.com/pulsar-edit/property-accessors/actions/workflows/ci.yaml/badge.svg)](https://github.com/pulsar-edit/property-accessors/actions/workflows/ci.yaml)
```
Comment (icecream17) [inline README.md:1]: was the CI badge intended to be part of the header?

otherwise, lgtm (looking through each commit is very helpful)

Comment (confused-Techie) [reply README.md:1]: Thanks a ton for looking at this! But yeah this was intentional, so that we can show the status of our GitHub Actions on the readme.

Comment (icecream17) [reply README.md:1]: <details open>
<summary>test</summary>

# Property Accessors Mixin [CI](https://github.com/pulsar-edit/property-accessors/actions/workflows/ci.yaml)

</details>

hmm, so like https://github.com/pulsar-edit/less-cache
(usually i see these below the header so got thrown there)

**Review by icecream17 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (confused-Techie) [reply README.md:1]: Yeah exactly, but it is a bit non-standard. But how they were done beforehand, so I thought to stick with it. Although wouldn't see any issues if we wanted to move them at all

**Comment by confused-Techie** [comment]
Thanks for approval! I'll go ahead and merge this one and create the new tag

**Pull request merged**